// evergreen-ui/commonjs/theme/src/default-theme/foundational-styles/colors.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _palette = _interopRequireDefault(require("./palette"));

var _scales = _interopRequireDefault(require("./scales"));

/**
 * This object are functional colors being used the default theme.
 * A required property is required by the default theme NOT by Evergreen itself.
 */
var _default = {
  /**
   * Available on Pane as `<Pane background="tint1" />`
   * @property {string} background.tint1 - Lightest tinted background color. Required property.
   * @property {string} background.tint2 - Slightly darker tinted background color. Required property.
   * @property {string} background.overlay - color used for the overlay
   */
  background: {
    tint1: _scales["default"].neutral.N1,
    tint2: _scales["default"].neutral.N2,
    overlay: _scales["default"].neutral.N7A,
    // Non required.
    yellowTint: _palette["default"].yellow.lightest,
    greenTint: _palette["default"].green.lightest,
    orangeTint: _palette["default"].orange.lightest,
    redTint: _palette["default"].red.lightest,
    blueTint: _palette["default"].blue.lightest,
    purpleTint: _palette["default"].purple.lightest,
    tealTint: _palette["default"].teal.lightest
  },

  /**
   * Available on Pane as `<Pane borderBottom borderRight="muted" />`
   * @property {string} text.default - Required property.
   * @property {string} text.muted - Slightly lighter color than default. Required property.
   */
  border: {
    "default": _scales["default"].neutral.N4,
    // Was BorderColors.muted in v3 and under.
    muted: _scales["default"].neutral.N3 // Was BorderColors.extraMuted in v3 and under

  },

  /**
   * Text colors available on Text as `<Text color="muted" />`.
   * @property {string} text.muted - Required property.
   * @property {string} text.default - Required property. Default for text.
   * @property {string} text.dark - Required property. Default for headings.
   */
  text: {
    muted: _scales["default"].neutral.N8,
    "default": _scales["default"].neutral.N9,
    dark: _scales["default"].neutral.N10,
    selected: _palette["default"].blue.base,
    // Intent.
    success: _palette["default"].green.dark,
    info: _palette["default"].blue.dark,
    danger: _palette["default"].red.dark,
    warning: _palette["default"].orange.dark
  },

  /**
   * Icon colors available on Icon.
   * @property {string} icon.default - Required property.
   * @property {string} icon.muted - Required property.
   * @property {string} icon.selected - Required property.
   */
  icon: {
    "default": _scales["default"].neutral.N8,
    muted: _scales["default"].neutral.N7,
    disabled: _scales["default"].neutral.N5A,
    selected: _palette["default"].blue.base,
    // Intent.
    success: _palette["default"].green.base,
    info: _palette["default"].blue.base,
    danger: _palette["default"].red.base,
    warning: _palette["default"].orange.base
  },

  /**
   * Used for Alerts and other (future) componentes that express intent.
   * @property {string} intent.none - Required property.
   * @property {string} intent.success - Required property.
   * @property {string} intent.danger - Required property.
   * @property {string} intent.warning - Required property.
   */
  intent: {
    none: _palette["default"].blue.base,
    success: _palette["default"].green.base,
    danger: _palette["default"].red.base,
    warning: _palette["default"].orange.base
  }
};
exports["default"] = _default;
