// evergreen-ui/esm/typography/src/Code.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { forwardRef, memo } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { useTheme } from '../../theme';
import Text from './Text';
var Code = memo(forwardRef(function Code(props, ref) {
  var theme = useTheme();

  var className = props.className,
      _props$appearance = props.appearance,
      appearance = _props$appearance === void 0 ? 'default' : _props$appearance,
      restProps = _objectWithoutProperties(props, ["className", "appearance"]);

  var _theme$getCodeProps = theme.getCodeProps(appearance),
      _theme$getCodeProps$c = _theme$getCodeProps.className,
      themedClassName = _theme$getCodeProps$c === void 0 ? '' : _theme$getCodeProps$c,
      themeProps = _objectWithoutProperties(_theme$getCodeProps, ["className"]);

  return React.createElement(Text, _extends({
    is: "code",
    ref: ref,
    className: cx(className, themedClassName),
    fontFamily: "mono"
  }, themeProps, restProps));
}));
Code.propTypes = _objectSpread(_objectSpread({}, Text.propTypes), {}, {
  /**
   * The appearance of the code.
   */
  appearance: PropTypes.oneOf(['default', 'minimal']),

  /**
   * Class name passed to the button.
   * Only use if you know what you are doing.
   */
  className: PropTypes.string
});
export default Code;
