// evergreen-ui/esm/typography/src/Heading.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { forwardRef, memo } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import { useTheme } from '../../theme';
var Heading = memo(forwardRef(function Heading(props, ref) {
  var theme = useTheme();

  var marginTop = props.marginTop,
      _props$size = props.size,
      size = _props$size === void 0 ? 500 : _props$size,
      restProps = _objectWithoutProperties(props, ["marginTop", "size"]);

  var _theme$getHeadingStyl = theme.getHeadingStyle(size),
      defaultMarginTop = _theme$getHeadingStyl.marginTop,
      headingStyle = _objectWithoutProperties(_theme$getHeadingStyl, ["marginTop"]);

  var finalMarginTop = marginTop;

  if (marginTop === 'default') {
    finalMarginTop = defaultMarginTop;
  }

  return React.createElement(Box, _extends({
    is: "h2",
    ref: ref,
    marginTop: finalMarginTop || 0,
    marginBottom: 0
  }, headingStyle, restProps));
}));
Heading.propTypes = _objectSpread(_objectSpread({}, Box.propTypes), {}, {
  /**
   * The size of the heading.
   */
  size: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900]),

  /**
   * Pass `default` to use the default margin top for that size.
   */
  marginTop: PropTypes.oneOfType([PropTypes.bool, PropTypes.number, PropTypes.string])
});
export default Heading;
