// evergreen-ui/commonjs/autocomplete/src/AutocompleteItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Option = _interopRequireDefault(require("../../select-menu/src/Option"));

var AutocompleteItem = (0, _react.memo)((0, _react.forwardRef)(function AutocompleteItem(props, ref) {
  var isHighlighted = props.isHighlighted,
      isSelected = props.isSelected,
      style = props.style,
      children = props.children,
      restProps = (0, _objectWithoutProperties2["default"])(props, ["isHighlighted", "isSelected", "style", "children"]);
  return _react["default"].createElement(_Option["default"], (0, _extends2["default"])({
    ref: ref,
    isHighlighted: isHighlighted,
    isSelected: isSelected,
    label: children,
    style: style
  }, restProps));
}));
AutocompleteItem.propTypes = {
  children: _propTypes["default"].node,
  style: _propTypes["default"].object,
  isSelected: _propTypes["default"].bool,
  isHighlighted: _propTypes["default"].bool
};
var _default = AutocompleteItem;
exports["default"] = _default;
