// evergreen-ui/commonjs/theme/src/default-theme/component-specific/getTabClassName.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _themer = require("../../../../themer");

var _memoizeClassName = _interopRequireDefault(require("../utils/memoizeClassName"));

var _scales = _interopRequireDefault(require("../foundational-styles/scales"));

var _shared = require("../shared");

/**
 * Disabled styles are all the same.
 */
var disabled = _shared.defaultControlStyles.disabled;

var defaultAppearance = _themer.Themer.createTabAppearance({
  base: {},
  hover: {
    backgroundColor: _scales["default"].neutral.N2A
  },
  focus: {
    boxShadow: "0 0 0 2px ".concat(_scales["default"].blue.B5A)
  },
  active: {
    backgroundColor: _scales["default"].blue.B3A,
    color: _scales["default"].blue.B9
  },
  disabled: disabled,
  current: {}
});
/**
 * Get the appearance of a `Tab`.
 * @param {string} appearance
 * @return {string} the appearance object.
 */


var getTabAppearance = function getTabAppearance() {
  return defaultAppearance;
};
/**
 * Get the className of a `Tab`.
 * @param {string} appearance
 * @return {string} the appearance class name.
 */


var _default = (0, _memoizeClassName["default"])(getTabAppearance);

exports["default"] = _default;
