// evergreen-ui/esm/ssr/src/extractStyles.js
import React from 'react';
import { extractStyles as boxExtractStyles } from 'ui-box';
import { renderStatic } from 'glamor/server';
export default function extractStyles() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

  var _boxExtractStyles = boxExtractStyles(),
      styles = _boxExtractStyles.styles,
      cache = _boxExtractStyles.cache;

  var _renderStatic = renderStatic(function () {
    return 'let glamor believe there is some html here';
  }),
      ids = _renderStatic.ids,
      css = _renderStatic.css;

  var evergreenCache = {
    uiBoxCache: cache,
    glamorIds: ids
  };
  var scriptProps = {
    type: 'application/json',
    id: 'evergreen-hydrate',
    dangerouslySetInnerHTML: {
      __html: JSON.stringify(evergreenCache)
    }
  };

  if (options.nonce) {
    scriptProps.nonce = options.nonce;
  }

  return {
    css: styles + '\n' + css,
    cache: evergreenCache,
    hydrationScript: React.createElement("script", scriptProps)
  };
}
