// evergreen-ui/commonjs/ssr/src/extractStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = extractStyles;

var _react = _interopRequireDefault(require("react"));

var _uiBox = require("ui-box");

var _server = require("glamor/server");

function extractStyles() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

  var _boxExtractStyles = (0, _uiBox.extractStyles)(),
      styles = _boxExtractStyles.styles,
      cache = _boxExtractStyles.cache;

  var _renderStatic = (0, _server.renderStatic)(function () {
    return 'let glamor believe there is some html here';
  }),
      ids = _renderStatic.ids,
      css = _renderStatic.css;

  var evergreenCache = {
    uiBoxCache: cache,
    glamorIds: ids
  };
  var scriptProps = {
    type: 'application/json',
    id: 'evergreen-hydrate',
    dangerouslySetInnerHTML: {
      __html: JSON.stringify(evergreenCache)
    }
  };

  if (options.nonce) {
    scriptProps.nonce = options.nonce;
  }

  return {
    css: styles + '\n' + css,
    cache: evergreenCache,
    hydrationScript: _react["default"].createElement("script", scriptProps)
  };
}
