// evergreen-ui/commonjs/menu/src/MenuOptionsGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _layers = require("../../layers");

var _typography = require("../../typography");

var _MenuOption = _interopRequireDefault(require("./MenuOption"));

var MenuOptionsGroup = (0, _react.memo)((0, _react.forwardRef)(function MenuOptionsGroup(props, ref) {
  var title = props.title,
      options = props.options,
      selected = props.selected,
      onChange = props.onChange;
  return _react["default"].createElement(_layers.Pane, {
    ref: ref,
    paddingY: 8
  }, title && _react["default"].createElement(_typography.Heading, {
    size: 100,
    marginLeft: 44,
    marginRight: 16,
    marginY: 8
  }, title), _react["default"].createElement(_layers.Pane, null, options.map(function (option) {
    return _react["default"].createElement(_MenuOption["default"], {
      key: option.value,
      isSelected: option.value === selected,
      onSelect: function onSelect() {
        return onChange(option.value);
      }
    }, option.label);
  })));
}));
MenuOptionsGroup.propTypes = {
  /**
   * Title of the menu group.
   */
  title: _propTypes["default"].node,

  /**
   * The current value of the option group.
   */
  selected: _propTypes["default"].any,

  /**
   * Function called when selection changes.
   */
  onChange: _propTypes["default"].func,

  /**
   * List of options rendered in the group.
   */
  options: _propTypes["default"].array
};
var _default = MenuOptionsGroup;
exports["default"] = _default;
