// evergreen-ui/commonjs/menu/src/MenuGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _layers = require("../../layers");

var _typography = require("../../typography");

var MenuGroup = (0, _react.memo)(function MenuGroup(props) {
  var title = props.title,
      children = props.children;
  return _react["default"].createElement(_layers.Pane, {
    paddingY: 8
  }, title && _react["default"].createElement(_typography.Heading, {
    size: 100,
    marginX: 16,
    marginY: 8
  }, title), children);
});
MenuGroup.propTypes = {
  /**
   * Title of the menu group.
   */
  title: _propTypes["default"].node,

  /**
   * The children of the menu group.
   */
  children: _propTypes["default"].node
};
var _default = MenuGroup;
exports["default"] = _default;
