// evergreen-ui/commonjs/table/src/EditableCellField.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _textarea = require("../../textarea");

var EditableCellField = (0, _react.memo)(function EditableCellField(props) {
  var getTargetRef = props.getTargetRef;

  var getTableBodyRef = function getTableBodyRef() {
    var ref = getTargetRef();
    if (!ref) return;

    while (ref) {
      var isTableBody = ref.hasAttribute('data-evergreen-table-body');

      if (isTableBody) {
        return ref;
      }

      if (ref.parentElement) {
        ref = ref.parentElement;
      } else {
        return null;
      }
    }

    return ref;
  };

  var latestAnimationFrame = (0, _react.useRef)();
  var textareaRef = (0, _react.useRef)();
  var tableBodyRef = (0, _react.useRef)();

  var _useState = (0, _react.useState)({
    top: 0,
    left: 0,
    height: 0,
    width: 0
  }),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      _useState2$ = _useState2[0],
      height = _useState2$.height,
      width = _useState2$.width,
      top = _useState2$.top,
      left = _useState2$.left,
      setDimensions = _useState2[1]; // Mirrors functionality of componentDidMount and componentWillUnmount.
  // Focus on mount


  (0, _react.useEffect)(function () {
    update();
    var requestId = requestAnimationFrame(function () {
      if (textareaRef.current) {
        textareaRef.current.focus();
      }
    });
    return function () {
      cancelAnimationFrame(requestId);

      if (latestAnimationFrame.current) {
        cancelAnimationFrame(latestAnimationFrame.current);
      }

      props.onCancel();
    };
  }, []);

  var update = function update() {
    var getTargetRef = props.getTargetRef;
    var targetRef = getTargetRef();
    if (!targetRef) return;
    tableBodyRef.current = getTableBodyRef();

    var _targetRef$getBoundin = targetRef.getBoundingClientRect(),
        targetLeft = _targetRef$getBoundin.left,
        targetTop = _targetRef$getBoundin.top,
        targetHeight = _targetRef$getBoundin.height,
        targetWidth = _targetRef$getBoundin.width;

    var calculatedTop;

    if (tableBodyRef.current) {
      var bounds = tableBodyRef.current.getBoundingClientRect();
      calculatedTop = Math.min(Math.max(targetTop, bounds.top), bounds.bottom - targetHeight);
    } else {
      calculatedTop = targetTop;
    }

    setDimensions({
      top: calculatedTop,
      left: targetLeft,
      height: targetHeight,
      width: targetWidth
    });
    latestAnimationFrame.current = requestAnimationFrame(function () {
      return update();
    });
  };

  var handleFocus = function handleFocus(e) {
    e.target.selectionStart = e.target.value.length;
  };

  var handleBlur = function handleBlur() {
    if (textareaRef.current) props.onChangeComplete(textareaRef.current.value);
  };

  var handleKeyDown = function handleKeyDown(e) {
    switch (e.key) {
      case 'Escape':
        props.onCancel();
        if (textareaRef.current) textareaRef.current.blur();
        break;

      case 'Enter':
        if (textareaRef.current) textareaRef.current.blur();
        e.preventDefault();
        break;

      case 'Tab':
        if (textareaRef.current) textareaRef.current.blur();
        break;

      default:
        break;
    }
  };

  var size = props.size,
      value = props.value,
      _props$minWidth = props.minWidth,
      minWidth = _props$minWidth === void 0 ? 80 : _props$minWidth,
      _props$minHeight = props.minHeight,
      minHeight = _props$minHeight === void 0 ? 40 : _props$minHeight,
      zIndex = props.zIndex;
  return _react["default"].createElement(_textarea.Textarea, {
    ref: textareaRef,
    onKeyDown: handleKeyDown,
    onBlur: handleBlur,
    onFocus: handleFocus,
    appearance: "editable-cell",
    size: size,
    style: {
      left: left,
      top: top,
      height: height,
      minHeight: Math.max(height, minHeight),
      width: width,
      minWidth: Math.max(width, minWidth),
      zIndex: zIndex
    },
    height: null,
    width: null,
    minHeight: null,
    position: "fixed",
    defaultValue: value
  });
});
EditableCellField.propTypes = {
  /**
   * Used as the defaultValue of the textarea.
   */
  value: _propTypes["default"].string.isRequired,

  /**
   * The z-index placed on the element.
   */
  zIndex: _propTypes["default"].number.isRequired,

  /**
   * Function to get the target ref of the parent.
   * Used to mirror the position.
   */
  getTargetRef: _propTypes["default"].func.isRequired,

  /**
   * Min width of the textarea.
   * The textarea can never be smaller than the cell.
   */
  minWidth: _propTypes["default"].number,

  /**
   * Min height of the textarea.
   * The textarea can never be smaller than the cell.
   */
  minHeight: _propTypes["default"].number,

  /**
   * Called when the textarea is blurred, pass the value back to the cell.
   */
  onChangeComplete: _propTypes["default"].func.isRequired,

  /**
   * Called when Escape is hit or componentWillUnmount.
   */
  onCancel: _propTypes["default"].func.isRequired,

  /**
   * Text size of the textarea.
   */
  size: _propTypes["default"].number
};
var _default = EditableCellField;
exports["default"] = _default;
