// evergreen-ui/esm/tooltip/src/TooltipStateless.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { memo, forwardRef } from 'react';
import PropTypes from 'prop-types';
import { Pane } from '../../layers';
import { Paragraph } from '../../typography';
import { useTheme } from '../../theme';
var TooltipStateless = memo(forwardRef(function TooltipStateless(props, ref) {
  var theme = useTheme();

  var children = props.children,
      appearance = props.appearance,
      restProps = _objectWithoutProperties(props, ["children", "appearance"]);

  var _theme$getTooltipProp = theme.getTooltipProps(appearance),
      color = _theme$getTooltipProp.color,
      themedProps = _objectWithoutProperties(_theme$getTooltipProp, ["color"]);

  var child;

  if (typeof children === 'string') {
    child = React.createElement(Paragraph, {
      color: color,
      size: 400
    }, children);
  } else {
    child = children;
  }

  return React.createElement(Pane, _extends({
    ref: ref,
    borderRadius: 3,
    paddingX: 8,
    paddingY: 4,
    maxWidth: 240
  }, themedProps, restProps), child);
}));
TooltipStateless.propTypes = {
  children: PropTypes.node,

  /**
   * The appearance of the tooltip.
   */
  appearance: PropTypes.oneOf(['default', 'card']).isRequired
};
export default TooltipStateless;
